File: Python/main_lists.py
from lists.linked_list import LinkedList
from lists.doubly_linked_list import DoublyLinkedList
from lists.circular_linked_list import CircularLinkedList

# list = LinkedList[int]()
# list = DoublyLinkedList[int]()
list = CircularLinkedList[int]()

if __name__ == '__main__':
    while True:
        print('\nMenu:')
        print('1  - Insert element in the beginning')
        print('2  - Insert element in the end')
        print('3  - Insert element (any position)')
        print('4  - Remove element in the beginning')
        print('5  - Remove element in the end')
        print('6  - Remove element (any)')
        print('7  - Get quantity de elements in the list')
        print('8  - Get element by index')
        print('9  - Get index by element')
        print('10 - Alterate element')
        print('11 - Print list')
        print('12 - Clear list')
        option = int(input('Option: '))

        if option == 1:
            el = int(input("\nType it the element to be insert: "))
            list.insert(0, el)

            print(f'\n{list}')

        elif option == 2:
            el = int(input("\nType it the element to be insert: "))
            list.append(el)

            print(f'\n{list}')

        elif option == 3:
            el = int(input("\nType it the element to be insert: "))
            index = int(input("Type it the new element's index: "))

            list.insert(index, el)

            print(f'\n{list}')

        elif option == 4:
            list.remove(list[0])

            if len(list):
                print(f'\n{list}')
            else:
                print("\nThe list it's empty!")

        elif option == 5:
            list.pop()

            if len(list):
                print(f'\n{list}')
            else:
                print("\nThe list it's empty!")

        elif option == 6:

            el = int(input('\nType it th element to be remove: '))
            list.remove(el)

            if len(list):
                print(f'\n{list}')
            else:
                print("\nThe list it's empty!")

        elif option == 7:
            if len(list):
                print(f"\nQuantity of elements in the list: {len(list)}")

            else:
                print("\nThe list it's empty!\n")

        elif option == 8:
            index = int(input("\nType it the element's index: "))
            el = list[index]

            print(f'\nElement in position {index}: {el}')

        elif option == 9:
            el = int(input("\nType it the element: "))
            index = list.index(el)

            print(f'\nElement in position {index}: {el}')

        elif option == 10:
            index = int(
                input("\nType it the index of the element to be alterate: "))
            el = int(input("Type it the new element: "))

            list[index] = el

            if list[index] == el:
                print("\nSuccessful alteration!\n")
                print(f"{list}")

            else:
                print("\nThere was an error in the alteration!")

        elif option == 11:
            if len(list):
                print(f'\n{list}')

            else:
                print("\nThe list it's empty!")

        elif option == 12:
            list.clear()

            if not len(list):
                print("\nThe list it's empty!")

        else:
            print("\nInvalid Option!")

File: TypeScript/main-lists.ts
import { questionInt } from 'readline-sync'
// import List from './data-structures/lists/linked-list'
// import List from './data-structures/lists/doubly-linked-list'
import List from './data-structures/lists/circular-linked-list'

let option: number, el: unknown, index: number
const list = new List()

do {
    option = questionInt(`
Menu:
1  - Insert element in the beginning
2  - Insert element in the end
3  - Insert element (any position)
4  - Remove element in the beginning
5  - Remove element in the end
6  - Remove element (any)
7  - Get quantity of elements in the list
8  - Get element by index
9  - Get index by element
10 - Alterate element
11 - Print list
12 - Clear list
Option: `)

    if (option === 1) {

        el = questionInt('\nType the element to be insert: ')
        list.insert(el)

        console.log(`\n${list.toString()}`)

    } else if (option === 2) {

        el = questionInt('\nType the element to be insert: ')
        list.push(el)

        console.log(`\n${list.toString()}`)

    } else if (option === 3) {

        index = questionInt('\nType the index of the new element: ')
        el = questionInt('Type the new element: ')

        if (list.insert(el, index)) {
            console.log(`\n${list.toString()}`)

        } else {
            console.log(`\nTh insert failed!`)

        }

    } else if (option === 4) {

        list.removeAt(0)

        if (list.length) {
            console.log(`\n${list.toString()}`)

        } else {
            console.log('\nThe list is empty!')
        }

    } else if (option === 5) {

        list.pop()
        
        if (list.length) {
            console.log(`\n${list.toString()}`)

        } else {
            console.log('\nThe list is empty!')
        }

    } else if (option === 6) {

        if (list.length) {
            el = questionInt('\nType the element to be remove: ')
            index = list.indexOf(el)

            if (index !== -1) {

                list.removeAt(index)

                if (list.length) {
                    console.log(`\n${list.toString()}`)

                } else {
                    console.log('\nThe list is empty!')
                }

            } else {
                console.log(`\n${el} is not in the list!`)
            }

        } else {
            console.log('\nThe list is empty!')

        }

    } else if (option === 7) {

        if (!list.length) {
            console.log('\nThe list is empty!')

        } else {
            console.log(`\nQuantity of elements in the list: ${list.length}`)
        }

    } else if (option === 8) {

        index = questionInt('\nType the new index: ')
        el = list.getElementAt(index)

        if (el !== undefined) {
            console.log(`\nElement in position ${index}: ${el}`)

        } else {
            console.log(`${el} is not in the list\n!`)
        }

    } else if (option === 9) {

        el = questionInt('\nType the element: ')
        index = list.indexOf(el)

        if (list.indexOf(el) !== -1) {
            console.log(`\nElement in position ${index}: ${el}`)

        } else {
            console.log(`${el} is not in the list\n!`)
        }

    } else if (option === 10) {

        index = questionInt('\nType the index of the element to be alterate: ')
        el = questionInt('Type the new element: ')

        list.setElementAt(el, index)

        if (list.getElementAt(index) === el) {
            console.log(`\Successful alteration!\n`)
            console.log(list.toString())

        } else {
            console.log(`\nThere was an error in the alteration!`)
        }

    } else if (option === 11) {

        if (!list.length) {
            console.log('\nThe list is empty!');

        } else {
            console.log(`\n${list.toString()}`)

        }

    } else if (option === 12) {

        list.clear()

        if (!list.length) {
            console.log('\nThe list is empty!')
        }

    } else {
        console.log('\nInvalid Option!')
    }

} while (1)